tgg/research/gnomad_tgps.py

```diff
@@ -52,111 +62,125 @@
-    pangenome.
+    Count the number of common (AF > 1%) variants per gnomAD sample.
 
-    :param str pangenome_ids_path: Cloud path for the text file containing pangenome
-    sample IDs.
-    :param str output_dir: Cloud path to output the result file.
-    :return: None; function writes .txt file to output directory.
+    Function produces counts only for gnomAD samples that are not part of the pangenome.
+
+    :param str pangenome_ids_path: Path to text file (stored in Google bucket)
+    containing pangenome sample IDs.
+    :param str output_dir: Path to a Google bucket for output file.
+    :return: None; function writes a text file to output bucket.
     """
-    # Get variant site frequency table.
+    # Get public gnomAD release sites HT.
     gnomad_freq_ht = release_sites().ht()
 
-    # Get gnomAD v3 data in split form.
-    mt = get_gnomad_v3_vds().variant_data
-    mt = hl.experimental.sparse_split_multi(mt)
-
     # Get gnomAD v3 metadata and filter to released samples.
     meta_ht = meta.ht()
     meta_ht = meta_ht.filter(meta_ht.release == True)
 
+    # Read gnomAD v3 raw data and split multi-allelics.
+    mt = get_gnomad_v3_vds().variant_data
+    mt = hl.experimental.sparse_split_multi(mt)
+
     # Use the filtered metadata to filter gnomAD v3 data.
     mt = mt.filter_cols(hl.is_defined(meta_ht[mt.col_key]))
     num_of_v3_samples = mt.count_cols()
-    logger.info("The number of samples in gnomAD v3 is: ", num_of_v3_samples)
+    logger.info("The number of samples in gnomAD v3 is: %i", num_of_v3_samples)
+    if num_of_v3_samples != NUM_OF_GNOMAD_SAMPLES:
+        raise DataException(
+            f"Found {num_of_v3_samples} samples in VDS but was expecting"
+            f" {NUM_OF_GNOMAD_SAMPLES} samples; please double check!"
+        )
 
     # Get pangenome sample IDs.
     pangenome_ids = get_pangenome_ids(pangenome_ids_path)
-    logger.info("The total number of samples in pangenome is ", hl.eval(hl.len(
+    logger.info("The total number of samples in pangenome is %i", hl.eval(hl.len(
         pangenome_ids)))
 
-    # Filter mt to two new matrix tables: one containing gnomAD samples not in
-    # pangenome, and one containing samples in pangenome.
+    # Filter mt to contain gnomAD samples not in pangenome.
     mt = mt.annotate_cols(s_in_pangenome=pangenome_ids.contains(mt.s))
-    pg_mt = mt.filter_cols(mt.s_in_pangenome)
     mt = mt.filter_cols(~mt.s_in_pangenome)
 
+    # Annotate mt with allele frequencies and filter out rare SNVs (AF lower than 1%).
+    # Also filter out sites where no samples had a variant call.
+    mt = mt.annotate_rows(freq=gnomad_freq_ht[mt.row_key].freq)
+    mt = mt.filter_rows(mt.freq.AF[0] > 0.01)
+    mt = mt.filter_rows(hl.agg.any(mt.GT.is_non_ref()))
+
+    # Read the HGDP+1KG subset matrix table and split multi-allelics.
+    pg_mt = hl.read_matrix_table(HGDP_SUBSET_MT)
+    pg_mt = hl.experimental.sparse_split_multi(pg_mt)
```

Comment:
```suggestion
    pg_mt = hl.experimental.sparse_split_multi(pg_mt)
    pg_mt = pg_mt.filter_rows(hl.len(pg_mt.alleles) == 1)
```
adding an additional filter to drop unnecessary rows earlier